### ISSUE TITLE
Kibana to Elasticsearch authentication failure (opendistro debian packages).

### ISSUE THREAD

**Issue by chaami** [opendistro-for-elasticsearch/community#65]
Hi,
First of all, thank you for providing debian packages...
I have used them to install the Opendistro Elasticsearch and Kibana flavors, but I am failing to make my them communicate (the operating system is Ubuntu 18.04).
In the elasticsearch security plugin `config` I had at first setup the `basic_internal_auth_domain` and `saml_auth_domain` authentication then reverted to simply using the `basic_internal_auth_domain` but with no better luck..
When starting the Kibana server, I end up with error messages like this one in the elasticsearch logs:
```
 Authentication finally failed for kibana from 127.0.0.1:39354
```
Which makes me think that the Kibana server is trying to connect as user `kibana` instead of `kibanaserver` as stated in the `/etc/kibana/kibana.yml`.
If that's the case, the failure makes sense as only `kibanauser` is set up in `internal_users.yml`.
I have confirmed that the configuration file is loaded as stating a faulty elasticsearch server there triggers errors.
I had added the Kibana configuration for SAML as I was trying to setup that authentication as it is explained in [the documentation](https://opendistro.github.io/for-elasticsearch-docs/docs/security/saml/#kibana-configuration).
Removing them didn't seem to have any effect.

Making curl requests as `kibanaserver` to list the installed plugins succeeds...
Making curl requests with a wrong password prevents from receiving the answer and pops a message indicating that the authentication did fail for `kibanaserver`

I am out of easy tricks...
Where can I set the user that shall be used or which password should be associated a kibana user to be created ?

Thanks and kind regards.

**Comment by allenyin96** [comment]
Hi @chaami,

Can you clarify if you were never able to make them communicate or did it stop working once you set up basic_internal_auth_domain?

Here're some default config for kibana.yml that should work:
```
opendistro_security.auth.type: "basicauth"
opendistro_security.basicauth.enabled: false
opendistro_security.multitenancy.enabled: true
opendistro_security.multitenancy.show_roles: true
opendistro_security.multitenancy.enable_filter: true
opendistro_security.multitenancy.tenants.enable_global: true
opendistro_security.multitenancy.tenants.enable_private: true

elasticsearch.hosts: ["https://localhost:9200"]
elasticsearch.username: "kibanaserver"
elasticsearch.password: "kibanaserver"
elasticsearch.ssl.verificationMode: none
```

If using the above didn't resolve your problem, can you please share your kibana.yml and elasticsearch.yml?

To answer your questions, internal_users.yml is where you set up users and their passwords.

**Comment by chaami** [comment]
Hi @allenyin96 ,
Thank you for posting !
I had failed to make Kibana and Elasticsearch actually communicate.
After long and tedious debugging sessions I managed to find the explanation...
A kibana.keystore was hanging in the Kibana directory from a previous installation that hadn't been cleaned-up properly...
The `elasticssearch.username` and `elasticsearch.password` were just being silently discarded by Kibana when trying to contact the Elasticsearch cluster during the healthcheck.
I had managed to reverse engineer the user and password in use and was going to report progress... and thanks to your post I went a little further and finally understood the origin of all this mess !
Jeez... this has been really painful !
Thanks again and sorry for the noise.
Kind regards.
